src/parser.rs

```diff
@@ -455,10 +512,25 @@ where
                     _ => (),
                 }
             }
-            _ => {
-                self.start_node(NODE_ERROR);
+            kind => {
+                let start = self.start_error_node();
                 self.bump();
-                self.finish_node();
+                let end = self.finish_error_node();
+                self.errors.push(ParseError::UnexpectedWanted(
+                    kind,
+                    TextRange::from_to(start, end),
+                    [
+                        TOKEN_PAREN_OPEN,
```

Comment:
Do you think this is warranted a helper function?

Reply:
I dont know if this makes sense here as you would need to pass the list of expected tokens anyways, thus it will not safe a lot of space, or make it more readable imho.